### PR TITLE
Added delegation events and modified undelegation command to the new …

### PR DESCRIPTION
**Pull request by ze97286**
…format

### CODE REVIEW

File: sources/vega/commands/v1/commands.proto [hunk 1]
```diff
@@ -128,10 +128,15 @@ message DelegateSubmission {
...
 
-// A command to submit an instruction to undelete stake from a node
-message UndelegateAtEpochEndSubmission {
-  // The ID for the node to undelegate from
+message UndelegateSubmission {
+  enum Method {
+    Now = 0;
+    AtEndOfEpoch =1;
+    InAnger = 2;
+    // more to come eventually
```
Comment (jeremyletang) [inline sources/vega/commands/v1/commands.proto:136]: I realise I wrote the sample fo this initially, but the protobuf linting we are using will require these enum variant to be all named in SNAKE_CASE_CAPITALISED, prefixed with the name of enum METHOD_*,  start part a variant name METHOD_UNSPECIFIED

File: commands/delegate_submission.go [hunk 2]
```diff
@@ -26,23 +26,23 @@ func checkDelegateSubmission(cmd *commandspb.DelegateSubmission) Errors {
...
 	if cmd == nil {
-		return errs.FinalAddForProperty("undelegateAtEpochEnd_submission", ErrIsRequired)
+		return errs.FinalAddForProperty("undelegate_submission", ErrIsRequired)
 	}
 
-	if cmd.Amount <= 0 {
-		errs.AddForProperty("undelegateAtEpochEnd_submission.amount", ErrIsRequired)
+	if cmd.Method > commandspb.UndelegateSubmission_InAnger {
+		errs.AddForProperty("undelegate_submission.method", ErrIsRequired)
 	}
```
Comment (jeremyletang) [inline commands/delegate_submission.go:42]: Should you not still verify the cmd.Amount here?

File: commands/delegate_submission.go [hunk 3]
```diff
@@ -26,23 +26,23 @@ func checkDelegateSubmission(cmd *commandspb.DelegateSubmission) Errors {
...
 	errs := NewErrors()
 
 	if cmd == nil {
-		return errs.FinalAddForProperty("undelegateAtEpochEnd_submission", ErrIsRequired)
+		return errs.FinalAddForProperty("undelegate_submission", ErrIsRequired)
 	}
 
-	if cmd.Amount <= 0 {
-		errs.AddForProperty("undelegateAtEpochEnd_submission.amount", ErrIsRequired)
+	if cmd.Method > commandspb.UndelegateSubmission_InAnger {
```
Comment (jeremyletang) [inline commands/delegate_submission.go:40]: you should probably do that instead:
```
if _, ok := UndelegateSubmission_Method_string[cmd.Method]; !ok || cmd.Method == commandspb.UndelegateSubmission_UNSPECIFIED {
    return MyERrror
}
```

File: commands/delegate_submission.go (hunk 2 above)
Comment (ze97286) [reply commands/delegate_submission.go:42]: verify how? if it's zero, it's taken as undelegate all. beyond that what can it be?

File: sources/vega/commands/v1/commands.proto (hunk 1 above)
Comment (ze97286) [reply sources/vega/commands/v1/commands.proto:136]: you're right I missed that

File: commands/delegate_submission.go (hunk 2 above)
Comment (jeremyletang) [reply commands/delegate_submission.go:42]: your absolutely right indeed.

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**